### ISSUE TITLE
compatibility with ignite metrics

### ISSUE THREAD

**Issue by benjamin-work** [dnouri/skorch#172]
If possible, add a scoring callback class that accepts an ignite metric as scoring function (see [here](https://github.com/pytorch/ignite/tree/master/ignite/metrics)).

**Comment by Nobody4591** [comment]
Hi @benjamin-work , can i start working on this?

**Comment by BenjaminBossan** [comment]
I completely forgot about this very old issue and I haven't kept up to date with the ignite development. I think one big advantage over sklearn metrics, which we already support, would be that ignite metrics are calculated on torch tensors, so they can for instance take advantage of GPU acceleration. Therefore, this could still be worth trying.

Maybe as a first step, you could try a simple metric like accuracy and create a prototype (no docs or tests necessary for that). If it looks like it could work, we can iterate from there without wasting too much time in case it's not feasible.

**Comment by Nobody4591** [comment]
sure, we can give it a try

**Comment by Nobody4591** [comment]
Hi @BenjaminBossan, I'm looking for contributing guide for skorch but can't find. Can you please point out?

**Comment by BenjaminBossan** [comment]
We don't have a separate contribution guide. But honestly, there isn't much to it. In the [installation instructions](https://skorch.readthedocs.io/en/stable/user/installation.html#from-source), there is a section on how to proceed if you want to contribute. So follow this, run pytest and pylint on your newly added code and you should be good.

Given your recent PR, I assume that you're familiar with the general approach of contributing to an open source project. I.e. you know how to fork and create a PR, you know that you should stick close to the style of the existing code, etc. If you don't, just ask.

For this specific problem, I would add that:

1. API-wise, it would be convenient for users if it were similar to that of [`EpochScoring`](https://github.com/skorch-dev/skorch/blob/923b3145361e0aa1c7ff2f7c467369db09e08009/skorch/callbacks/scoring.py#L282), though the implementation will likely be very different.
2. I think it makes sense to start a new module for ignite metrics inside the `skorch/callbacks/` directory. Similarly, there should be a new test module, but that's for later.

The rest we can figure out on the go, which is why it's good to start with a proof of concept.

**Comment by Nobody4591** [comment]
Thanks!
I thinks it's enough to start with.
